### PR TITLE
Print tracebacks by default when running regression tests

### PR DESCRIPTION
**Pull request by shivupa**
This is a draft PR to add the traceback flag to pytest as mentioned [here](https://github.com/cclib/cclib/pull/945#issuecomment-701054430) .

### CODE REVIEW

**Comment by shivupa** [comment]
@berquist I am confused about this now. I think running pytest without the `--terse` flag gives what I would expect to happen with the `--traceback` flag. 

`python -m pytest -v  --capture=no --cov=cclib --cov-report=term --cov-report=html  test` vs `python -m pytest -v  --capture=no --cov=cclib --cov-report=term --cov-report=html  --terse test`

On another note the reason the traceback flag currently errors is because the `test_all` function in `test/test_data.py` doesn't accept a traceback flag. It does accept terse and silent flags which I think behave similarly. Regression tests do accept a traceback flag.

File: travis/run_pytest.bash [hunk 1]
```diff
@@ -5,8 +5,8 @@
 
 set -euxo pipefail
 
-python -m pytest -v --capture=no --cov=cclib --cov-report=term --cov-report=html --terse test
+python -m pytest -v --traceback --capture=no --cov=cclib --cov-report=term --cov-report=html --terse test
```
Comment (berquist) [inline travis/run_pytest.bash:8]: I would stick this at the end as an attempt to differentiate between `pytest` flags and cclib flags.

File: test/conftest.py [hunk 2]
```diff
@@ -56,6 +57,7 @@ def pytest_generate_tests(metafunc):
         metafunc.parametrize("modules", [{p: all_modules[p] for p in module_names}])
         metafunc.parametrize("terse", [metafunc.config.getoption("--terse")])
         metafunc.parametrize("silent", [metafunc.config.getoption("--silent")])
+        metafunc.parametrize("traceback", [metafunc.config.getoption("--traceback")])
```
Comment (berquist) [inline test/conftest.py:60]: This actually needs to go under a new block, `if metafunc.function.__name__ == "test_regressions"` and I think it'll be `opt_traceback`: https://github.com/cclib/cclib/blob/67be86feb8d97eb123c268b86323afec008d6dc5/test/regression.py#L3294
and you'll probably get the cryptic "default argument" error I was getting.

**Comment by berquist** [comment]
We agreed that a better solution is to remove the option and change the default of `opt_traceback` to `True`.

**Review by berquist — APPROVED** [review]

**Pull request merged**